R/estimateDiversity.R

```diff
@@ -104,31 +107,44 @@ setGeneric("estimateRichness",signature = c("x"),
            function(x, ...)
                standardGeneric("estimateRichness"))
 
+#' @rdname estimateDiversity
+#' @export
+setGeneric("estimateCoverage",signature = c("x"),
+           function(x, ...)
+               standardGeneric("estimateCoverage"))
 
 #' @rdname estimateDiversity
 #' @export
 setMethod("estimateDiversity", signature = c(x = "SummarizedExperiment"),
-    function(x, abund_values = "counts",
-             index = c("shannon","simpson","inv_simpson", "richness", "chao1",
-                       "ACE"),
-             name = index, ..., BPPARAM = SerialParam()){
-        # input check
-        index<- match.arg(index, several.ok = TRUE)
-        if(!.is_non_empty_character(name) || length(name) != length(index)){
-            stop("'name' must be a non-empty character value and have the ",
-                 "same length than 'index'.",
-                 call. = FALSE)
-        }
-        .check_abund_values(abund_values, x)
-        .require_package("vegan")
-        #
-        dvrsts <- BiocParallel::bplapply(index,
-                                         .run_dvrsty,
-                                         x = x,
-                                         mat = assay(x, abund_values),
-                                         BPPARAM = BPPARAM, ...)
-        .add_values_to_colData(x, dvrsts, name)
-    }
+          function(x, abund_values = "counts",
+                   index = c("shannon","simpson","inv_simpson", "richness", "chao1",
+                             "ACE", "coverage"),
+                   name = index, threshold = NULL, ..., BPPARAM = SerialParam()){
+              # input check
+              index<- match.arg(index, several.ok = TRUE)
+              if(!.is_non_empty_character(name) || length(name) != length(index)){
+                  stop("'name' must be a non-empty character value and have the ",
+                       "same length than 'index'.",
+                       call. = FALSE)
+              }
+              .check_abund_values(abund_values, x)
+              .require_package("vegan")
+
+              # If the index is coverage, and threshold is not specified, or it is under 0 or over 1
+              if( ("coverage" %in% index) && (is.null(threshold) || threshold < 0 || threshold > 1) ){
+                  stop("'threshold' must be specified when calculating 'coverage' index.
+                 'threshold' must be a numeric value between 0-1.",
+                       call. = FALSE)
+              }
+              #
+              dvrsts <- BiocParallel::bplapply(index,
+                                               .run_dvrsty,
+                                               x = x,
+                                               mat = assay(x, abund_values),
+                                               threshold = threshold,
+                                               BPPARAM = BPPARAM, ...)
+              .add_values_to_colData(x, dvrsts, name)
+          }
```

Comment:
I think just changed the indentation. Please revert this and at every other place

Reply:
I copy pasted the content of file, and it seems that it wasn't smart move, because it messed the indentation. Now it is fixed.